pkg/logs/config/integration_config.go

```diff
@@ -220,23 +220,32 @@ func ValidateProcessingRules(rules []LogsProcessingRule) error {
 			return fmt.Errorf("LogsAgent misconfigured: type must be set for log processing rule `%s`", rule.Name)
 		default:
 			return fmt.Errorf("LogsAgent misconfigured: type %s is unsupported for log processing rule `%s`", rule.Type, rule.Name)
-		}
+		}		
 	}
 	return nil
 }
 
 // CompileProcessingRules compiles all processing rules regular expression
-func CompileProcessingRules(rules []LogsProcessingRule) {
+func CompileProcessingRules(rules []LogsProcessingRule) error {
 	for i, rule := range rules {
-		switch rule.Type {
-		case ExcludeAtMatch, IncludeAtMatch:
-			rules[i].Reg = regexp.MustCompile(rule.Pattern)
-		case MaskSequences:
-			rules[i].Reg = regexp.MustCompile(rule.Pattern)
-			rules[i].ReplacePlaceholderBytes = []byte(rule.ReplacePlaceholder)
-		case MultiLine:
-			rules[i].Reg = regexp.MustCompile("^" + rule.Pattern)
-		default:
+		if rule.Pattern != "" {
+			re, err := regexp.Compile(rule.Pattern)
+			if err != nil {
+				fmt.Errorf("LogsAgent misconfigured: pattern %s is unsupported for log processing rule: `%s`", rule.Pattern, rule.Name)
```

Comment:
nits, we don't start error message with uppercase letter

Reply:
👍 Was following what I saw in the other `fmt.Errorf()` in the code.  Should I change all the others to start with lower case, too?